custom_dataset.py

```diff
@@ -0,0 +1,408 @@
+        self._GRAYSCALE_OUT = config.getboolean('IMAGE', 'GRAYSCALE_OUT')
+
+        # Get traning and testing spec
+        self._TRAIN_BATCH_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                              'TRAIN_BATCH_SIZE')
+        self._TEST_BATCH_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                             'TEST_BATCH_SIZE')
+        self._SHUFFLE_BUFFER_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                                 'SHUFFLE_BUFFER_SIZE')
+        self._PREFETCH_BUFFER_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                                  'PREFETCH_BUFFER_SIZE')
+
+        # Get data augmentation spec
+        self._RANDOM_ROTATE = config.getboolean('DATA_AUG', 'RANDOM_ROTATE')
+        self._ROTATE_STDDEV = config.getfloat('DATA_AUG', 'ROTATE_STDDEV')
+
+        self._RANDOM_ZOOM = config.getboolean('DATA_AUG', 'RANDOM_ZOOM')
+        self._ZOOM_PERCENT = config.getfloat('DATA_AUG', 'ZOOM_PERCENT')
+        self._ZOOM_STDDEV = config.getfloat('DATA_AUG', 'ZOOM_STDDEV')
+
+        self._RESIZE = config.getboolean('DATA_AUG', 'RESIZE')
+        self._RESIZE_HEIGHT = config.getint('DATA_AUG', 'RESIZE_HEIGHT')
+        self._RESIZE_WIDTH = config.getint('DATA_AUG', 'RESIZE_WIDTH')
+
+        # Label conversion
+        self._CLASS_NAMES = [] # Class names in the same order as in label file
+        with open(self._LABEL_FILE) as f:
+            for line in f:
+                code_point = line.split('\n')[0]
+                self._CLASS_NAMES.append(code_point)
+        self._NUM_CLASSES = len(self._CLASS_NAMES) # Number of classes
+
+    def _get_label(self, file_path):
+        """Given a path to label file. Expect label file to be in the format
+        generated by 'source/generate_source_file.py'.
+
+        Args:
+            file_path: Str, path to label file.
+
+        Returns:
+            if self._ONE_HOT:
+                label: tf.Tensor, one hot encoding of label.
+            else:
+                idx: tf.Tensor, label index.
+        """
+        # Convert path to file name
+        file_name = tf.strings.split(file_path, os.path.sep)[-1]
+        # Derive label from file name
+        class_name = tf.strings.split(file_name, '_')[0]
+        idx = tf.reduce_min(tf.where(tf.equal(self._CLASS_NAMES, class_name)))
+        if self.ONE_HOT:
+            label = tf.one_hot(idx, self._NUM_CLASSES)
+            return label
+        else:
+            return idx
+
+    def _decode_img(self, img):
+        """Decode image and convert to tf.Tensor.
+
+        Args:
+            img: tf.Tensor, image of type String.
+
+        Returns:
+            img: tf.Tensor, image of type float32.
+        """
+        # Convert compressed string to a 3D uint8 tensor
+        img = tf.io.decode_png(img)
+        # Convert data type to float between 0 and 1
+        img = tf.image.convert_image_dtype(img, tf.float32)
```

Comment:
just
return tf.image.convert_image_dtype(img, tf.float32)
no need to assign it to img and then return img